nvflare/fuel/sec/authz.py

```diff
@@ -22,35 +23,27 @@
 
 
 class Person(object):
-    def __init__(self, name: str, org: str, role):
+    def __init__(self, name: str, org: str, role: str):
         self.name = _normalize_str(name)
         self.org = _normalize_str(org)
-        self.roles = []
-        if isinstance(role, str):
-            self.roles.append(_normalize_str(role))
-        elif isinstance(role, list):
-            if len(role) <= 0:
-                raise TypeError("roles not specified - it must be a list of strings")
-
-            for r in role:
-                if not isinstance(r, str):
-                    raise TypeError(f"role value must be a str but got {type(r)}")
-                self.roles.append(_normalize_str(r))
-        else:
-            raise TypeError(f"role must be a str or list of str but got {type(role)}")
+        self.role = _normalize_str(role)
 
     def __str__(self):
-        return f"{self.name}:{self.org}:{self.roles[0]}"
+        return f"{self.name}:{self.org}:{self.role}"
 
 
 class AuthzContext(object):
     def __init__(self, right: str, user: Person, submitter: Person = None):
         """Base class to contain context data for authorization."""
+        if not isinstance(user, Person):
+            raise ValueError(f"user need to be of type Person but got {type(user)}")
+        if submitter and not isinstance(submitter, Person):
+            raise ValueError(f"submitter need to be of type Person but got {type(submitter)}")
         self.right = right
         self.user = user
         self.submitter = submitter
         self.attrs = {}
-        if not submitter:
+        if submitter is None:
```

Comment:
The original one is better. It can handle empty string.

Reply:
I added the checks in L40-41, so we should be good.